core/src/main/java/io/grpc/transport/HttpUtil.java

```diff
@@ -121,5 +121,90 @@ public static Status httpStatusToGrpcStatus(int httpStatusCode) {
     return Status.UNKNOWN;
   }
 
+  /**
+   * All error codes identified by the HTTP/2 spec.
+   */
+  public enum Http2Error {
+    NO_ERROR(0x0, Status.INTERNAL),
+    PROTOCOL_ERROR(0x1, Status.INTERNAL),
+    INTERNAL_ERROR(0x2, Status.INTERNAL),
+    FLOW_CONTROL_ERROR(0x3, Status.INTERNAL),
+    SETTINGS_TIMEOUT(0x4, Status.INTERNAL),
+    STREAM_CLOSED(0x5, Status.INTERNAL),
+    FRAME_SIZE_ERROR(0x6, Status.INTERNAL),
+    REFUSED_STREAM(0x7, Status.UNAVAILABLE),
+    CANCEL(0x8, Status.CANCELLED),
+    COMPRESSION_ERROR(0x9, Status.INTERNAL),
+    CONNECT_ERROR(0xA, Status.INTERNAL),
+    ENHANCE_YOUR_CALM(0xB, Status.RESOURCE_EXHAUSTED, "Bandwidth exhausted"),
```

Comment:
Not sure if this will work.  The default constructor is already doing withDescription with the HTTP/2 error code so that it's clear that the error is a result of an inbound HTTP/2 error.  We could use augmentDescription, in which case the error will be something like "HTTP/2 error code: X Bandwidth exhausted".  Is that what you're thinking?